source/darwin/commoncrypto_aes.c

```diff
@@ -355,6 +353,37 @@ struct aws_symmetric_cipher *aws_aes_ctr_256_new_impl(
     return &cc_cipher->cipher_base;
 }
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
if your macro above is correct, you should never be warned of deprecated declarations.

Reply:
Discussed offline, we need this for the runtime check.